torch/distributed/checkpoint/default_planner.py

```diff
@@ -178,6 +179,7 @@ def set_up_planner(
         metadata: Metadata,
         is_coordinator: bool,
     ) -> None:
+        state_dict = _initialize_meta_tensors(state_dict)
```

Comment:
This won't work. What if the state_dict is nested?

You can write something like
```
def init_meta(v):
      ...
      if isinstance(value, DTensor):
           ...
      elif isinstance(value, ShardedTensor):
           ...
      else:
           return torch.empty_like(value, device=device)
  
  state_dict = tree_map_only(torch.Tensor, init_meta)        
 ```